test/testframe.c

```diff
@@ -11,368 +11,504 @@
+                    ret_value = FAIL;
+                    goto done;
+                }
             }
             else {
-                TestUsage();
-                exit(EXIT_FAILURE);
+                ret_value = FAIL;
+                goto done;
             }
         }
         else if (((strcmp(*argv, "-exclude") == 0) || (strcmp(*argv, "-x") == 0))) {
             if (argc > 0) {
                 --argc;
                 ++argv;
-                SetTest(*argv, SKIPTEST);
+
+                if (SetTest(*argv, SKIPTEST) < 0) {
+                    ret_value = FAIL;
+                    goto done;
+                }
             }
             else {
-                TestUsage();
-                exit(EXIT_FAILURE);
+                ret_value = FAIL;
+                goto done;
             }
         }
         else if (((strcmp(*argv, "-begin") == 0) || (strcmp(*argv, "-b") == 0))) {
             if (argc > 0) {
                 --argc;
                 ++argv;
-                SetTest(*argv, BEGINTEST);
+
+                if (SetTest(*argv, BEGINTEST) < 0) {
+                    ret_value = FAIL;
+                    goto done;
+                }
             }
             else {
-                TestUsage();
-                exit(EXIT_FAILURE);
+                ret_value = FAIL;
+                goto done;
             }
         }
         else if (((strcmp(*argv, "-only") == 0) || (strcmp(*argv, "-o") == 0))) {
             if (argc > 0) {
-                unsigned Loop;
-
                 --argc;
                 ++argv;
 
-                /* Skip all tests, then activate only one. */
-                if (!skipped_all) {
-                    for (Loop = 0; Loop < Index; Loop++)
-                        Test[Loop].SkipFlag = 1;
-                    skipped_all = true;
-                } /* end if */
-                SetTest(*argv, ONLYTEST);
+                if (SetTest(*argv, ONLYTEST) < 0) {
+                    ret_value = FAIL;
+                    goto done;
+                }
             }
             else {
-                TestUsage();
-                exit(EXIT_FAILURE);
+                ret_value = FAIL;
+                goto done;
             }
         }
         else if ((strcmp(*argv, "-summary") == 0) || (strcmp(*argv, "-s") == 0))
-            Summary = 1;
-        else if (strcmp(*argv, "-enable-error-stack") == 0)
-            enable_error_stack = 1;
+            TestDoSummary_g = true;
+        else if (strcmp(*argv, "-disable-error-stack") == 0) {
+            TestEnableErrorStack = false;
+        }
         else if ((strcmp(*argv, "-help") == 0) || (strcmp(*argv, "-h") == 0)) {
-            TestUsage();
+            TestUsage(stdout);
             exit(EXIT_SUCCESS);
         }
         else if ((strcmp(*argv, "-cleanoff") == 0) || (strcmp(*argv, "-c") == 0))
             SetTestNoCleanup();
+        else if ((strcmp(*argv, "-maxthreads") == 0) || (strcmp(*argv, "-t") == 0)) {
```

Comment:
Added a new `-maxthreads` command-line option for testframe.c tests for use with multi-threaded tests in the future.